apps/opik-backend/src/main/java/com/comet/opik/domain/llmproviders/LlmProviderAnthropicMapper.java

```diff
@@ -0,0 +1,122 @@
+import dev.ai4j.openai4j.chat.SystemMessage;
+import dev.ai4j.openai4j.chat.UserMessage;
+import dev.ai4j.openai4j.shared.Usage;
+import dev.langchain4j.model.anthropic.internal.api.AnthropicContent;
+import dev.langchain4j.model.anthropic.internal.api.AnthropicCreateMessageRequest;
+import dev.langchain4j.model.anthropic.internal.api.AnthropicCreateMessageResponse;
+import dev.langchain4j.model.anthropic.internal.api.AnthropicMessage;
+import dev.langchain4j.model.anthropic.internal.api.AnthropicMessageContent;
+import dev.langchain4j.model.anthropic.internal.api.AnthropicRole;
+import dev.langchain4j.model.anthropic.internal.api.AnthropicTextContent;
+import dev.langchain4j.model.anthropic.internal.api.AnthropicUsage;
+import jakarta.ws.rs.BadRequestException;
+import lombok.NonNull;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+import org.mapstruct.factory.Mappers;
+
+import java.util.List;
+
+@Mapper
+public interface LlmProviderAnthropicMapper {
+    LlmProviderAnthropicMapper INSTANCE = Mappers.getMapper(LlmProviderAnthropicMapper.class);
+
+    @Mapping(source = "response", target = "choices", qualifiedByName = "mapToChoices")
+    @Mapping(source = "usage", target = "usage", qualifiedByName = "mapToUsage")
+    ChatCompletionResponse toResponse(@NonNull AnthropicCreateMessageResponse response);
+
+    @Mapping(source = "content", target = "message")
+    @Mapping(source = "response.stopReason", target = "finishReason")
+    ChatCompletionChoice toChoice(@NonNull AnthropicContent content, @NonNull AnthropicCreateMessageResponse response);
+
+    @Mapping(source = "text", target = "content")
+    AssistantMessage toAssistantMessage(@NonNull AnthropicContent content);
+
+    @Mapping(expression = "java(request.model())", target = "model")
+    @Mapping(expression = "java(request.stream())", target = "stream")
+    @Mapping(expression = "java(request.temperature())", target = "temperature")
+    @Mapping(expression = "java(request.topP())", target = "topP")
+    @Mapping(expression = "java(request.stop())", target = "stopSequences")
+    @Mapping(expression = "java(request.maxCompletionTokens())", target = "maxTokens")
+    @Mapping(source = "request", target = "messages", qualifiedByName = "mapToMessages")
+    @Mapping(source = "request", target = "system", qualifiedByName = "mapToSystemMessages")
+    AnthropicCreateMessageRequest toCreateMessageRequest(@NonNull ChatCompletionRequest request);
+
+    @Named("mapToChoices")
+    default List<ChatCompletionChoice> mapToChoices(@NonNull AnthropicCreateMessageResponse response) {
+        if (response.content == null || response.content.isEmpty()) {
+            return List.of();
+        }
+        return response.content.stream().map(content -> toChoice(content, response)).toList();
+    }
+
+    @Named("mapToUsage")
+    default Usage mapToUsage(AnthropicUsage usage) {
+        if (usage == null) {
+            return null;
+        }
```

Comment:
Can we use validations to avoid this or use Optional instead?

Reply:
I cannot add `@NonNull` validations because it actually might be `null`. As for changing to `Optional`, this is an anti pattern, I prefer to avoid.

Reply:
No, the anti-pattern is receiving Optional as a param. I'm referring to returning Optional since you must receive a null

Reply:
I thought that's what you meant. Anyway, I can't get map struct to work when returning an `Optional`.